spring-data-azure-documentdb/src/main/java/com/microsoft/azure/spring/data/documentdb/config/AbstractDocumentDbConfiguration.java

```diff
@@ -0,0 +1,43 @@
+package com.microsoft.azure.spring.data.documentdb.config;
+
+import com.microsoft.azure.documentdb.DocumentClient;
+import com.microsoft.azure.spring.data.documentdb.DocumentDbFactory;
+import com.microsoft.azure.spring.data.documentdb.core.DocumentDbTemplate;
+import com.microsoft.azure.spring.data.documentdb.core.convert.DocumentDbConverter;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+public abstract class AbstractDocumentDbConfiguration extends DocumentDbConfigurationSupport {
+
+    @Value("${azure.documentdb.databasename}")
```

Comment:
Better to use @ConfigurationProperties as suggested by spring to use in spring 4+ versions , though @Value will work and there is nothing wrong here